sdk/tutorials/upgrade-eoa-to-smart-account.md

```diff
@@ -0,0 +1,495 @@
+  const [transactionHash, setTransactionHash] = useState<string | null>(null);
+  const [statusError, setStatusError] = useState<string | null>(null);
+// add-next-line
++ const [statusLoading, setStatusLoading] = useState(false);
+
+// add-start
++ const handleGetCallsStatus = async () => {
++   if (!data?.id) return;
++
++   setStatusLoading(true);
++   setStatusError(null);
++
++   try {
++     const status = await getCallsStatus(config, { id: data.id });
++     console.log("Transaction status:", status);
++
++     if (
++       status.status === "success" &&
++       status.receipts?.[0]?.transactionHash
++     ) {
++       setTransactionHash(status.receipts[0].transactionHash);
++     } else if (status.status === "failure") {
++       setStatusError("Transaction failed");
++     }
++   } catch (err) {
++     console.error("Error getting call status:", err);
++     setStatusError(
++       err instanceof Error ? err.message : "Failed to get transaction status"
++     );
++   } finally {
++     setStatusLoading(false);
++   }
++ };
+// add-end
+  // ...
+```
+
+Then, create a button to check the batch transaction status:
+
+```tsx title="page.tsx"
+// ...
+
+  {/* Transaction state */}
+  // ...
+
+// add-start
++ {/* Check transaction status button */}
++ {data && (
++   <button
++     className={`w-full rounded-lg border border-solid px-6 py-3 font-medium transition-colors ${
++       statusLoading
++         ? "bg-gray-100 text-gray-400 border-gray-300 cursor-not-allowed"
++         : "bg-purple-50 hover:bg-purple-100 text-purple-700 border-purple-300 cursor-pointer"
++     }`}
++     onClick={handleGetCallsStatus}
++     disabled={statusLoading || !data.id}
++   >
++     {statusLoading
++       ? "Checking Status..."
++       : "Check Transaction Status"}
++   </button>
++ )}
+// add-end
+
+// add-start
++ {/* Status error */}
++ {statusError && (
++   <div className="bg-red-50 border border-red-200 rounded-lg p-4 mt-4">
++     <div className="text-red-700 font-medium">Status Check Error</div>
++     <div className="text-sm text-red-600 mt-1">{statusError}</div>
++   </div>
++ )}
+// add-end
+
+// add-start
++ {/* Transaction hash */}
++ {transactionHash && (
++   <div className="bg-blue-50 border border-blue-200 rounded-lg p-4 mt-4">
++     <div className="text-blue-700 font-medium mb-2">
++       Transaction Confirmed!
++     </div>
++     <div className="text-sm">
++       <a
++         href={`https://sepolia.etherscan.io/tx/${transactionHash}`}
++         target="_blank"
++         rel="noopener noreferrer"
++         className="text-blue-600 hover:text-blue-800 hover:underline break-all"
++       >
++         View on Etherscan: {transactionHash}
++       </a>
++     </div>
++   </div>
++ )}
+// add-end
+// ...
+```
+
+In the development server, when you send a new batch transaction, if the transaction is successful, the success state and the **Check Transaction Status** button displays.
```

Comment:
Something like, "If a batch transaction succeeds in the development server, the success state and the Check Transaction Status button appear."?